### PR TITLE
Load k8s client-go auth plugins

### PR DESCRIPTION
**Pull request by ibrasho**
If `kubeconfig` is configured to use an auth provider (`gcp`, `oauth` ... etc) you won't be able to connect to the cluster and will get the dreaded error message: 
`No Auth Provider found for name "AUTH_PROVIDER"`.

The change is this PR ensure all default auth plugins for `k8s.io/client-go` are loaded as per [this document](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/client-go/examples/README.md).

### CODE REVIEW

**Comment by ibrasho** [comment]
I'm not sure what's the project policy on go modules vendoring? I vendored everything just incase.

**Comment by ibrasho** [comment]
The only real change in this PR is this one line: https://github.com/digitalocean/clusterlint/pull/51/files#diff-18c42ff6e08ecf5257441b61ea0ae06cR27

Everything else is go modules magic.

File: kube/objects.go [hunk 1]
```diff
@@ -24,6 +24,7 @@ import (
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/client-go/kubernetes"
+	_ "k8s.io/client-go/plugin/pkg/client/auth"
```
Comment (varshavaradarajan) [inline kube/objects.go]: Can you please add a comment above this import statement to show indicate why it is unused?

**Comment by varshavaradarajan** [comment]
@ibrasho - there are no changes to `go.mod` that's been checked in. Was that missed out accidentally?

**Comment by jcodybaker** [comment]
I've confirmed this works on a GKE cluster.

File: kube/objects.go (hunk 1 above)
Comment (ibrasho) [reply kube/objects.go]:
```go
	// Load client-go authentication plugins
	_ "k8s.io/client-go/plugin/pkg/client/auth"
```

Does this clarify the reason enough? I thought about adding the link as a reference, but I'm not sure.

**Pull request merged**